### PR TITLE
Validate circuit vote before attempting consensus

### PR DESCRIPTION
**Pull request by peterschwarz**
Like circuit proposals, validate the contents of a circuit vote before it can be submitted to the consensus system.  This allows the client to detect rejected  votes at submission time, instead of at consensus time, long after the HTTP request has completed.

Additionally

- fix a typo
- ignore IO errors when scanning the path for pandoc

To test:

1. Build gameroom with standard flags
2. Create Alice, with her key in the registry as normal
3. Propose a gameroom
4. Create Charlie, with a new key
5. Accept the invite
6. An error message should be displayed immediately

### CODE REVIEW

File: cli/build.rs [hunk 1]
```diff
@@ -31,11 +31,25 @@ fn main() -> Result<(), BuildError> {
...
+                continue;
+            }
+        };
+
+        for entry in entries {
+            let entry = match entry {
+                Ok(entry) => entry,
+                Err(err) => {
+                    // skip an entry in the path that cannot be read.
+                    println!("Unable read entry in {}: {}", path, err);
```
Comment (jsmitchell) [inline cli/build.rs]: s/Unable read/Unable to read/

File: libsplinter/src/admin/service/shared.rs [hunk 2]
```diff
@@ -664,6 +664,38 @@ impl AdminServiceShared {
...
+                let proposal_vote = payload.get_circuit_proposal_vote();
+
+                // validate vote proposal
+                // check that the circuit proposal exists
+                let circuit_proposal = self
+                    .get_proposal(proposal_vote.get_circuit_id())
+                    .map_err(|err| {
+                        ServiceError::UnableToHandleMessage(Box::new(
+                            AdminSharedError::ValidationFailed(format!(
+                                "error occured when trying to get proposal {}",
```
Comment (jsmitchell) [inline libsplinter/src/admin/service/shared.rs]: s/occured/occurred/

Comment (peterschwarz) [reply libsplinter/src/admin/service/shared.rs]: Damn, I fixed that typo in one place, but reproduced it else where...

Fixed

File: cli/build.rs (hunk 1 above)
Comment (peterschwarz) [reply cli/build.rs]: Fixed

File: libsplinter/src/admin/service/shared.rs (hunk 2 above)
Comment (jsmitchell) [reply libsplinter/src/admin/service/shared.rs]: s/else where/elsewhere/

File: libsplinter/src/admin/service/shared.rs [hunk 3]
```diff
@@ -664,6 +664,38 @@ impl AdminServiceShared {
                 self.propose_circuit(payload, "local".to_string())
             }
             CircuitManagementPayload_Action::CIRCUIT_PROPOSAL_VOTE => {
+                let proposal_vote = payload.get_circuit_proposal_vote();
```
Comment (agunde406) [inline libsplinter/src/admin/service/shared.rs]: Can you add a log statement somewhere so we can see why the vote failed, I don't see any indication in the logs (maybe we should add one for circuit proposal too?)

Comment (peterschwarz) [reply libsplinter/src/admin/service/shared.rs]: Added as a new commit (at the submit handler layer).

**Comment by ltseeley** [comment]
Tested locally

**Pull request merged**